armoryengine/WalletUtils.py

```diff
@@ -319,27 +319,25 @@ def detectHighestUsedIndex(self):
 ## Wallet Data Classes
 ################################################################################
 
-class WalletEntry(object):
+class WalletListEntry(object):
    """Represents a single wallet entry with all its properties."""
 
-   def __init__(self, walletId, path, filename, displayName, staged, 
-                stateName, isLegacy, isEncrypted, isReady, entry):
-      self.walletId = walletId
-      self.path = path
-      self.filename = filename
-      self.displayName = displayName
-      self.staged = staged
-      self.stateName = stateName
-      self.isLegacy = isLegacy
-      self.isEncrypted = isEncrypted
-      self.isReady = isReady
-      self.entry = entry  # Keep reference for bridge operations
-
-class WalletListData(object):
-   """Container for processed wallet list data ready for UI rendering."""
-
-   def __init__(self, wallets):
-      self.wallets = wallets  # List of WalletEntry objects
+   def __init__(self, entry):
+      self.walletId = entry.walletId
+      self.path = entry.path
+      self.staged = entry.staged or False
+
+      self.stateName = str(entry.state).lower()
+      self.isLegacy = (self.stateName == 'legacy')
+      self.isEncrypted = (self.stateName == 'encrypted')
+      self.isReady = (self.stateName == 'ready')
+
+      self.filename = os.path.basename(self.path) if self.path else ''
+      self.displayName = self.walletId if self.walletId else os.path.splitext(self.filename)[0]
```

Comment:
You should be carrying the full path and rendering the isolated filename on demand instead.